test/e2e/garbage_collector.go

```diff
@@ -72,6 +106,39 @@ func newOwnerRC(f *framework.Framework, name string) *v1.ReplicationController {
 	}
 }
 
+// verifyRemainingDeploymentsAndReplicaSets verifies if the number of the remaining deployment
+// and rs are deploymentNum and rsNum. It returns error if the
+// communication with the API server fails.
+func verifyRemainingDeploymentsAndReplicaSets(f *framework.Framework, clientSet clientset.Interface, deployment *v1beta1.Deployment, deploymentNum, rsNum int) (bool, error) {
+	rs, err := clientSet.Extensions().ReplicaSets(f.Namespace.Name).List(v1.ListOptions{})
+	if err != nil {
+		return false, fmt.Errorf("Failed to list rs: %v", err)
+	}
+
+	// Count the ReplicaSets which belong to the Deleted Deployment
+	controllerRef := controller.GetControllerOf(deployment.ObjectMeta)
+	rsNumFound := 0
+	for _, replicaSet := range rs.Items {
+		if controllerRef != nil && controllerRef.UID == replicaSet.ObjectMeta.UID {
```

Comment:
This looks wrong. It should check if the replicaset's controllerRef is pointing to the deployment.